### PR TITLE
Carry #998: Use vndr tool for vendoring

### PR DESCRIPTION
**Pull request by dqminh**
This carries #998 and replaces godep with vndr.

Also remove usage of symlink GOPATH. Since our README mentions that you should build it inside GOPATH, i think its a reasonable to assume that you dont need to create a tmp GOPATH.

### CODE REVIEW

**Comment by dqminh** [comment]
ping @hqhq @crosbymichael @mrunalp @LK4D4

**Comment by cyphar** [comment]
/me sad that we have to drop the `GOPATH` tricks. I've always hated how Go handles `GOPATH`... :cry:

**Comment by cyphar** [comment]
@dqminh Can you remove the `Godeps` line from `.gitignore`?

**Comment by cyphar** [comment]
@dqminh Can you remove the `Godeps` line from `.gitignore`?

LGTM otherwise.

**Comment by dqminh** [comment]
> Can you remove the Godeps line from .gitignore?

Done.

> /me sad that we have to drop the GOPATH tricks. I've always hated how Go handles GOPATH...

haha yes i can understand, but i feel like symlinking is kinda worse :disappointed:

**Comment by cyphar** [comment]
LGTM -- let's agree to disagree on symlinking. :wink:

File: Makefile [hunk 1]
```diff
@@ -28,21 +26,21 @@ SHELL := $(shell command -v bash 2>/dev/null)
...
 all: runc recvtty
 
 recvtty: contrib/cmd/recvtty/recvtty
 
-contrib/cmd/recvtty/recvtty: $(SOURCES) | $(RUNC_LINK)
+contrib/cmd/recvtty/recvtty: $(SOURCES)
 	go build -i -ldflags "-X main.gitCommit=${COMMIT} -X main.version=${VERSION}" -tags "$(BUILDTAGS)" -o contrib/cmd/recvtty/recvtty ./contrib/cmd/recvtty
 
-static: $(SOURCES) | $(RUNC_LINK)
+static: $(SOURCES) .GOPATH/.ok
```
Comment (hqhq) [inline Makefile]: Where did you build this `.GOPATH/.ok`?

**Comment by hqhq** [comment]
I hate we have to put code under GOPATH, but I think symlink

File: Makefile (hunk 1 above)
Comment (dqminh) [inline Makefile]: removed. Story is that at work we have a makefile template exactly to do these kind of symlink trick ( if someone wants symlink ) https://github.com/cloudflare/hellogopher. And so i tried to integrate it here in some form. But then i feel like symlink isnt necessary here, so remove it.

**Comment by hqhq** [comment]
LGTM
We can close https://github.com/opencontainers/runc/issues/1235 after it's merged.

**Comment by cyphar** [comment]
LGTM

**Comment by cyphar** [comment]
:heart:

**Pull request merged**